hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoGetAccountInfoHandler.java

```diff
@@ -341,14 +341,13 @@ private StakingInfo getStakingInfo(
         final var stakingInfo =
                 StakingInfo.newBuilder().declineReward(account.declineReward()).stakedToMe(account.stakedToMe());
 
-        final var stakedNum = account.stakedNumber();
-        if (stakedNum < 0) {
-            // Staked num for a node is (-nodeId -1)
-            stakingInfo.stakedNodeId(-stakedNum - 1);
+        final var stakedNode = account.stakedNodeId();
+        final var stakedAccount = account.stakedAccountId();
+        if (stakedNode != null) {
```

Comment:
should this just be `account.hasStakedNodeId()`? same for line 349 `account.hasStakedAccountId()`?

Reply:
Yes, the null check and .hasXXX() are the same. Since I will need to get the values and fill them in stakingInfo, I did this way. If I ever need to debug, I can see the values.